modules/docker/inspect.go

```diff
@@ -0,0 +1,214 @@
+package docker
+
+import (
+	"encoding/json"
+	"github.com/gruntwork-io/terratest/modules/shell"
+	"github.com/stretchr/testify/require"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+// ContainerInspect defines the output of the Inspect method, with the options returned by 'docker inspect'
+// converted into a more friendly and testable interface
+type ContainerInspect struct {
+	// ID of the inspected container
+	ID       string
+
+	// Name of the inspected container
+	Name     string
+
+	// time.Time that the container was created
+	Created  time.Time
+
+	// String representing the container's status
+	Status   string
+
+	// Whether the container is currently running or not
+	Running  bool
+
+	// Container's exit code
+	ExitCode uint8
+
+	// String with the container's error message, if there is any
+	Error    string
+
+	// Ports exposed by the container
+	Ports    []Port
+
+	// Volume bindings made to the container
+	Binds    []VolumeBind
+}
+
+// Port represents a single port mapping exported by the container
+type Port struct {
+	HostPort      uint16
+	ContainerPort uint16
+	Protocol      string
+}
+
+// VolumeBind represents a single volume binding made to the container
+type VolumeBind struct {
+	Source      string
+	Destination string
+}
+
+// inspectOutput defines options that will be returned by 'docker inspect', in JSON format.
+// Not all options are included here, only the ones that we might need
+type inspectOutput struct {
+	Id      string
+	Created string
+	Name    string
+	State   struct {
+		Status   string
+		Running  bool
+		ExitCode uint8
+		Error    string
+	}
+	NetworkSettings struct {
+		Ports map[string][]struct {
+			HostIp   string
+			HostPort string
+		}
+	}
+	HostConfig struct {
+		Binds []string
+	}
+}
+
+// Inspect runs the 'docker inspect {container id}' command and returns a ContainerInspect
+// struct, converted from the output JSON, along with any errors
+func Inspect(t *testing.T, id string) *ContainerInspect {
+	out, err := InspectE(t, id)
+	require.NoError(t, err)
+
+	return out
+}
+
+// InspectE runs the 'docker inspect {container id}' command and returns a ContainerInspect
+// struct, converted from the output JSON, along with any errors
+func InspectE(t *testing.T, id string) (*ContainerInspect, error) {
+	cmd := shell.Command{
+		Command: "docker",
+		Args:    []string{"container", "inspect", id},
+	}
+
+	out, err := shell.RunCommandAndGetOutputE(t, cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	var containers []inspectOutput
+	err = json.Unmarshal([]byte(out), &containers)
+	if err != nil {
+		return nil, err
+	}
+
+	container := containers[0]
```

Comment:
Probably best to check `len(containers) > 0`. Not sure it's likely to happen, but better to check than panic at test time.

Reply:
True, by removing the check I accidentally removed a safe check for index out of range. I've added it back, returning an error it `containers` is empty.